src/XtensaOptimize.cpp

```diff
@@ -952,11 +990,18 @@ class MatchXtensaPatterns : public IRGraphMutator {
             // that they generate.
             internal_assert(op->args.size() == 3);
             return mutate(lower_lerp(op->type, op->args[0], op->args[1], op->args[2], target));
-        } else if (op->is_intrinsic(Call::absd) && op->type.is_vector() && op->type.is_uint() && (op->type.bits() == 16)) {
+        } else if (op->is_intrinsic(Call::absd) && op->type.is_vector() && op->type.is_uint()) {
             internal_assert(op->args.size() == 2);
-            return Call::make(op->type, "halide_xtensa_absd_i16",
-                              {mutate(op->args[0]), mutate(op->args[1])},
-                              Call::PureExtern);
+
+            if (op->type.bits() == 16) {
+                return Call::make(op->type, "halide_xtensa_absd_i16",
+                                  {mutate(op->args[0]), mutate(op->args[1])},
+                                  Call::PureExtern);
+            } else if (op->type.bits() == 8) {
+                return Call::make(op->type, "halide_xtensa_absd_u8",
+                                  {mutate(op->args[0]), mutate(op->args[1])},
+                                  Call::PureExtern);
+            }
```

Comment:
What if neither is true -- does the fall-through case handle it correctly? Or should this produce a compilation error?